src/bls.cpp

```diff
@@ -100,6 +100,14 @@ void BLS::HashPubKeys(bn_t* output, size_t numOutputs,
     CheckRelicErrors();
 }
 
+PublicKey BLS::DHKeyExchange(const PrivateKey& privKey, const PublicKey& pubKey) {
+    AssertInitialized();
+    if (!privKey.keydata) {
+        throw std::string("keydata not initialized");
+    }
+    return pubKey.Exp(*privKey.keydata);
```

Comment:
CheckRelicErrors after the exponentiation? I think we should do this in every function just in case